packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -288,6 +306,7 @@ class RouteSettings {
   /// Creates data used to construct routes.
   const RouteSettings({
     this.name,
+    this.accessibilityName,
```

Comment:
The place where you create the settings won't ever know what the right name should be. This should be in the route's widget tree, not in the route settings.